plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobContainer.java

```diff
@@ -248,11 +249,16 @@ public DeleteResult delete() throws IOException {
                     return s3Object.key();
                 }).collect(Collectors.toList());
 
-                if (!listObjectsResponseIterator.hasNext()) {
+                if (!listObjectsResponseIterator.hasNext() && softDelete == false) {
```

Comment:
Why are we not marking to delete the key path when softDelete is true?

Reply:
we can't delete the path when the objects in it are still existing .